lib/superset/guest_token.rb

```diff
@@ -65,5 +71,18 @@ def access_token
     def authenticator
       @authenticator ||= Superset::Authenticator.new(credentials)
     end
+
+    def rls_filter_clause
+      "#{site_clause} AND #{programme_clause}"
+    end
+
+    def site_clause
+      return true if current_user_sites.empty?
+      "site_id = ANY(ARRAY[#{current_user_sites}])" 
```

Comment:
❓ Is returning `true` intentional as to skip the clause?

Reply:
Yes, when accessible_site_ids is empty, the user must see records from all sites. Hence to skip that site_id check, we are setting it as true

Reply:
@soundarya-mv @msxavi
 
This Gem is now a public gem so should not have any application specific code in it.
If we will decided to go this direction .. we should over write this class in the Neptune codebase

Reply:
@jbat , yes I wanted to test this approach first, hence added it for now. Now I have tested the approach. Will update it to be generic.